channel/Monster3GPP38901.m

```diff
@@ -243,132 +243,41 @@ function propagateWaveforms(obj, Stations, Users, Mode)
 			% The channel is reciprocal in terms of received power, thus the path
 			% loss is extracted from channel conditions provided by
 			%
-			switch mode
-				case 'downlink'
-					lossdB = obj.computePathLoss(Station, User, Station.Tx.Freq);
-					EIRPdBm = Station.Tx.getEIRPdBm(Station.Position, User.Position);
-					receivedPower = EIRPdBm-lossdB+User.Rx.getLoss(Station.Position, User.Position); %dBm
-				case 'uplink'
-					lossdB = obj.computePathLoss(Station, User, User.Tx.Freq);
-					EIRPdBm = User.Tx.getEIRPdBm;
-					receivedPower = EIRPdBm-lossdB-Station.Rx.NoiseFigure; %dBm
-			end
-			receivedPowerWatt = 10^((receivedPower-30)/10);
-		end
-
-		function [receivedPower, receivedPowerWatt] = computeLinkBudgetMatrix(obj, Station, User, mode, sampleGrid)
-			% computeLinkBudgetMatrix
-			%
-			% :param obj:
-			% :param Station:
-			% :param User:
-			% :param mode:
-			% :param sampleGrid:
-			% :returns receivedPower:
-			% :returns receivedPowerWatt:
-			% 
-			% Matrix edition: Create 2d an 3d distances from sampleGrid.
-			% TODO: reformat this function to work with computeLinkBudget
-			
-
-			
-			[X, Y] = meshgrid(sampleGrid(1,:), sampleGrid(2,:));
-			d2 = arrayfun(@(x,y) obj.Channel.getDistance(Station.Position(1:2),[x y]), X, Y);
-			d3 = arrayfun(@(x,y) obj.Channel.getDistance(Station.Position(1:3),[x y User.Position(3)]), X, Y);
-			EIRPdBm = arrayfun(@(x,y) Station.Tx.getEIRPdBm(Station.Position, [x y]), X, Y);
-			DownlinkUeLoss = arrayfun(@(x,y) User.Rx.getLoss(Station.Position, [x y]), X, Y);
+			if isempty(varargin)
```

Comment:
More comments should be added here. Or use of a variable name that makes sense. The switch is fine, but it should be really obvious why this switch is necessary. One solution could be making an argument-based varargin. So, for instance, it could look like `...Station, User, mode, 'grid', grid)` . Then have some check in the beginning of the function that looks for 'grid'. Based on that, do the switch.